### PR TITLE
Tequila implementation

### PR DESCRIPTION
**Pull request by maasencioh**
- Update initial documentation for setting up the project
- Documentation over the tequila process
- Implementation of addon for tequila authentication

### CODE REVIEW

**Comment by maasencioh** [comment]
@targos @Xstoudi
Is it possible to have a value associated to an open session? As you can see in the [documentation](https://github.com/mylims/mylims/tree/docs/addons/tequila) the normal way to get the session is through a cookie. If I'm  not wrong the session is being handled differently, but we will still need to store a key to an associated session on the Tequila server (step 6 on the protocol).

**Comment by targos** [comment]
To store something in the session, use `session.put(key, value)`. To read it, `session.get('key')`
https://preview.adonisjs.com/guides/http/sessions#using-sessions

Please prefix your key with `tequila.` (name of the addon)

File: README.md [hunk 1]
```diff
@@ -35,6 +35,14 @@ node scripts/init-db.mjs
 npm run dev
 ```
 
+If it's the first time, don't forget to copy or create the `.env` file and is necessary to run the migration command
+
+```shell
+cp .env.example
```
Comment (targos) [inline README.md]:
```suggestion
cp .env.example .env
```

**Comment by Xstoudi** [comment]
LGTM when change suggested by Michael will be commited.

**Comment by maasencioh** [comment]
Please don't merge yet, is a draft and all the process stills needs validation

**Comment by maasencioh** [comment]
Hello guys, do you know how to test the full authentication process? Especially because we have a redirection and this is not clear to me.

**Comment by maasencioh** [comment]
@targos @Xstoudi this is now working, could you please take a look?
image

File: addons/tequila/Schemas/AuthMethods.graphql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+type AuthMethods {
```
Comment (Xstoudi) [inline addons/tequila/Schemas/AuthMethods.graphql]: Won't be necessary when #94 will be merged.

Comment (targos) [reply addons/tequila/Schemas/AuthMethods.graphql]: it is merged now

**Pull request merged**